client/mm/exchange_adaptor_test.go

```diff
@@ -0,0 +1,2943 @@
+	}
+
+	type test struct {
+		name               string
+		isDynamicSwapper   map[uint32]bool
+		balances           map[uint32]uint64
+		multiTrade         *core.MultiTradeForm
+		multiTradeResponse []*core.Order
+		wantErr            bool
+		postTradeBalances  map[uint32]*botBalance
+		updatesAndBalances []*updateAndBalances
+	}
+
+	tests := []*test{
+		{
+			name: "non dynamic swapper, sell",
+			balances: map[uint32]uint64{
+				42: 1e8,
+				0:  1e8,
+			},
+			multiTrade: &core.MultiTradeForm{
+				Host:  host,
+				Sell:  true,
+				Base:  42,
+				Quote: 0,
+				Placements: []*core.QtyRate{
+					{
+						Qty:  5e6,
+						Rate: 5e7,
+					},
+					{
+						Qty:  5e6,
+						Rate: 6e7,
+					},
+				},
+			},
+			multiTradeResponse: []*core.Order{
+				{
+					Host:      host,
+					BaseID:    42,
+					QuoteID:   0,
+					Sell:      true,
+					LockedAmt: 5e6 + 2000,
+					Status:    order.OrderStatusBooked,
+					ID:        orderIDs[0][:],
+				},
+				{
+					Host:      host,
+					BaseID:    42,
+					QuoteID:   0,
+					Sell:      true,
+					LockedAmt: 5e6 + 2000,
+					Status:    order.OrderStatusBooked,
+					ID:        orderIDs[1][:],
+				},
+			},
+			postTradeBalances: map[uint32]*botBalance{
+				42: {
+					Available: 1e8 - (5e6+2000)*2,
+					Locked:    (5e6 + 2000) * 2,
+				},
+				0: {
+					Available: 1e8,
+				},
+			},
+			updatesAndBalances: []*updateAndBalances{
+				// First order has a match and sends a swap tx
+				{
+					walletTxUpdates: []*asset.WalletTransaction{
+						{
+							Type:         asset.Swap,
+							ID:           walletTxIDs[0][:],
+							BalanceDelta: -2e6,
+							Fees:         1000,
+						},
+					},
+					note: &core.OrderNote{
+						Order: &core.Order{
+							Host:      host,
+							BaseID:    42,
+							QuoteID:   0,
+							Sell:      true,
+							LockedAmt: 3e6 + 1000,
+							Status:    order.OrderStatusBooked,
+							ID:        orderIDs[0][:],
+							Matches: []*core.Match{
+								{
+									Swap: &core.Coin{
+										ID: walletTxIDs[0][:],
+									},
+								},
+							},
+						},
+					},
+					balances: map[uint32]*botBalance{
+						42: {
+							Available: 1e8 - (5e6+2000)*2,
+							Locked:    5e6 + 2000 + 3e6 + 1000,
+						},
+						0: {
+							Available: 1e8,
+						},
+					},
+				},
+				// Second order has a match and sends swap tx
+				{
+					walletTxUpdates: []*asset.WalletTransaction{
+						{
+							Type:         asset.Swap,
+							ID:           walletTxIDs[1][:],
+							BalanceDelta: -3e6,
+							Fees:         1000,
+						},
+					},
+					note: &core.OrderNote{
+						Order: &core.Order{
+							Host:      host,
+							BaseID:    42,
+							QuoteID:   0,
+							Sell:      true,
+							LockedAmt: 2e6 + 1000,
+							Status:    order.OrderStatusBooked,
+							ID:        orderIDs[1][:],
+							Matches: []*core.Match{
+								{
+									Swap: &core.Coin{
+										ID: walletTxIDs[1][:],
+									},
+								},
+							},
+						},
```

Comment:
Please try to identify places where you can use variables and helper function for populating test data. It's hard to identify the parameters being tested.